torchgeo/datasets/glacier_mapping_alps.py

```diff
@@ -0,0 +1,419 @@
+        checksum: bool = False,
+    ) -> None:
+        """Initialize the dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            cv_iter: one of 1, 2, 3, 4, 5 (for the five-fold geographical cross-validation scheme)
+            version: one of "small" or "large" (controls the sampling overlap)
+            bands: the Sentinel-2 bands to use as input (default: RGB + NIR + SWIR)
+            extra_features: additional features to include (default: None; see the class attribute for the available)
+            transforms: a function/transform that takes input sample and its target as entry and returns a transformed
+             version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        self.root = Path(root)
+        self.split = split
+        self.cv_iter = cv_iter
+        self.version = version
+        self.bands = bands
+        self.extra_features = extra_features
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        # sanity checks
+        assert split in (
+            'train',
+            'val',
+            'test',
+        ), f'Split {split} not found in the available splits'
+        assert cv_iter in range(
+            1, 6
+        ), f'Cross-validation iteration {cv_iter} has to be in [1, 2, 3, 4, 5]'
+        assert version in (
+            'small',
+            'large',
+        ), f'Version {version} not found in the available versions'
+        for band in bands:
+            assert (
+                band in self.all_bands
+            ), f'Band {band} not found in the available Sentinel-2 bands'
+        if self.extra_features:
+            for feature in self.extra_features:
+                assert (
+                    feature in self.extra_features_all
+                ), f'Feature {feature} not found in the available extra features'
+
+        # set the local file paths
+        label = f'dataset_{version}'
+        self.fp_archive = self.root / f'{label}.tar.gz'
+        self.dir_patches = self.root / label
+        self.fp_splits_csv = self.root / 'splits.csv'
+
+        # get the corresponding urls and checksums
+        self.url_dataset = self.download_metadata[label]['url']
+        self.md5_dataset = self.download_metadata[label]['checksum']
+        self.url_csv_splits = self.download_metadata['splits_csv']['url']
+        self.md5_csv_splits = self.download_metadata['splits_csv']['checksum']
+
+        self._verify()
+        self._prepare_files()
+
+    def __len__(self) -> int:
+        """The length of the dataset.
+
+        It returns the number of patches in the dataset.
+        """
+        return len(self.fp_patches)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
```

Comment:
The doc string should also include "Args" and "Returns" explicitly, for the formatting check for example: https://github.com/microsoft/torchgeo/blob/5cca8e7336fc582ddc81a86e05d50b505ff74a07/torchgeo/datasets/caffe.py#L142